processor/src/test/java/org/mapstruct/ap/test/collection/map/AnnotatedMapMappingTest.java

```diff
@@ -0,0 +1,112 @@
+
+    @ProcessorTest
+    public void shouldContainTwoKeyMappingMapsInConstructor() {
+        generatedSource.forMapper( AnnotatedSourceTargetMapper.class )
+            .content()
+            .contains(
+                "private final Map<String, String> stringMapToStringMapUsingTargetParameterKeyMappings;" )
+            .contains(
+                "private final Map<String, String> stringMapToStringMapUsingTargetParameterAndReturnKeyMappings;" )
+            .contains( "private final Map<String, String> stringMapWithMappingAnnotationsKeyMappings;" )
+            .contains(
+                "private final Map<String, String> stringMapWithMappingAnnotationsAndDifferentNameKeyMappings;" )
+            .contains(
+                "private final Map<String, String> stringLongMapToStringIntegerMapWithMappingAnnotationsKeyMappings;" )
+            .doesNotContain( "private final Map<String, String> stringMapWithInvalidMappingAnnotationsKeyMappings;" )
+            .doesNotContain( "private final Map<String, String> longMapWithMappingAnnotationsKeyMappings;" )
+            .contains( expectedConstructor() );
+    }
+
+    private static String expectedConstructor() {
+        return "public AnnotatedSourceTargetMapperImpl() {" +
+            lineSeparator() +
+            constructorInitOf( "stringMapToStringMapUsingTargetParameterKeyMappings" ) +
+            lineSeparator() +
+            constructorInitOf( "stringMapToStringMapUsingTargetParameterAndReturnKeyMappings" ) +
+            lineSeparator() +
+            constructorInitOf( "stringMapWithMappingAnnotationsKeyMappings" ) +
+            lineSeparator() +
+            constructorInitOf( "stringMapWithMappingAnnotationsAndDifferentNameKeyMappings" ) +
+            lineSeparator() +
+            constructorInitOf( "stringLongMapToStringIntegerMapWithMappingAnnotationsKeyMappings" ) +
+            "    }";
+    }
+
+    private static String constructorInitOf(String parameterName) {
+        return
+            "        this." + parameterName + " = new java.util.HashMap<>();" +
+                lineSeparator() +
+                "        this." + parameterName + ".put( \"sourceKey\", \"targetKey\" );" +
+                lineSeparator() +
+                "        this." + parameterName + ".put( \"nonExistentSourceKey\", \"otherTargetKey\" );" +
+                lineSeparator();
+    }
+
+    @ProcessorTest
+    public void shouldMapBasedOnKeyMappings() {
+        Map<String, String> values = new HashMap<>();
+        values.put( "sourceKey", "value" );
+        values.put( "unmappedKey", "otherValue" );
+
+        Map<String, String> target = AnnotatedSourceTargetMapper.INSTANCE.stringMapWithMappingAnnotations( values );
+
+        assertThat( target ).isNotNull()
+            .hasSize( 2 )
+            .contains(
+                entry( "targetKey", "value" ),
+                entry( "unmappedKey", "otherValue" )
+            );
```

Comment:
we can remove `isNotNull` and `hasSize` if we use `containsOnly` instead of `contains`